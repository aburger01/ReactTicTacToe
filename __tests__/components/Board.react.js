import React from 'react';
import renderer from 'react-test-renderer';
import Board from '../../src/components/Board.react';
import Square from '../components/Square.react';

test('Does the Board component render correctly?', () => {

  const testBoard = ["X", "O", null, "X", "O", null, "X", null, null];
  const testWinners = [true, false, false, true, false, false, true, false, false];

  const tree = renderer.create(
    <Board squares={testBoard} winners={testWinners} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
