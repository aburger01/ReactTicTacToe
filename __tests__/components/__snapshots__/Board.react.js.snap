// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does the Board component render correctly? 1`] = `
<div>
  <div
    className="board-row"
  >
    <button
      className="square winner"
      onClick={[Function]}
    >
      X
    </button>
    <button
      className="square"
      onClick={[Function]}
    >
      O
    </button>
    <button
      className="square"
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <button
      className="square winner"
      onClick={[Function]}
    >
      X
    </button>
    <button
      className="square"
      onClick={[Function]}
    >
      O
    </button>
    <button
      className="square"
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <button
      className="square winner"
      onClick={[Function]}
    >
      X
    </button>
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`Does the Square component render correctly? 1`] = `
<button
  className="square"
  onClick={undefined}
>
  X
</button>
`;

exports[`Does the Squares class name change if winner is true? 1`] = `
<button
  className="square winner"
  onClick={undefined}
/>
`;
